File: scripts/sync_readme_version.js
const fs = require('fs');
const path = require('path');

const repoRoot = path.resolve(__dirname, '..');
const packageJsonPath = path.join(repoRoot, 'package.json');
const readmePath = path.join(repoRoot, 'README.md');

const pkg = JSON.parse(fs.readFileSync(packageJsonPath, 'utf8'));
const version = pkg.version || '';
const shortVersion = version.split('.').slice(0, 2).join('.');

const readme = fs.readFileSync(readmePath, 'utf8');
const updated = readme
  .replace(/__VERSION_SHORT__/g, shortVersion)
  .replace(/__VERSION_FULL__/g, version);

if (updated !== readme) {
  fs.writeFileSync(readmePath, updated);
  console.log(`README version synced to ${shortVersion} (package.json: ${version})`);
} else {
  console.log(`README already uses version ${shortVersion}`);
}

File: scripts/sync_readme_version.py
#!/usr/bin/env python3
import json
from pathlib import Path

repo_root = Path(__file__).resolve().parent.parent
package_path = repo_root / 'package.json'
readme_path = repo_root / 'README.md'

with package_path.open('r', encoding='utf-8') as f:
    pkg = json.load(f)

version = pkg.get('version', '')

readme = readme_path.read_text(encoding='utf-8')
updated = readme.replace('__VERSION_SHORT__', version).replace('__VERSION_FULL__', version)

if updated != readme:
    readme_path.write_text(updated, encoding='utf-8')
    print(f'README version synced to {version} (package.json: {version})')
else:
    print(f'README already uses version {version}')
